js/src/setup-mc/setup-stepper/setup-free-listings/shipping-rate/shipping-rate-setup/useShippingRatesWithSavedSuggestions.js

```diff
@@ -0,0 +1,148 @@
+/**
+ * External dependencies
+ */
+import { useState, useEffect, useCallback, useRef } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import useDispatchCoreNotices from '.~/hooks/useDispatchCoreNotices';
+import useShippingRates from '.~/hooks/useShippingRates';
+import getCountriesPriceArray from './getCountriesPriceArray';
+import { useAppDispatch } from '.~/data';
+import useShippingRatesSuggestions from './useShippingRatesSuggestions';
+
+/**
+ * Convert shipping rates suggestions to aggregated shipping rates
+ * that are ready to be saved.
+ *
+ * @param {Array<import('.~/data/actions').ShippingRate>} suggestions Shipping rate suggestions.
+ * @return {Array<import('.~/data/actions').AggregatedShippingRate>} Aggregated shipping rates.
+ */
+const convertSuggestionsToAggregatedShippingRates = ( suggestions ) => {
+	const countriesPriceArray = getCountriesPriceArray( suggestions );
+	const values = countriesPriceArray.map( ( el ) => ( {
+		countryCodes: el.countries,
+		currency: el.currency,
+		rate: el.price,
+	} ) );
```

Comment:
💅 

Duplicated code.

WDTY of creating a testable function like `mapToShippingRates`

Reply:
@puntope , do you mean it is duplicated with the lines 51-55 in `useShippingRatesSuggestions` https://github.com/woocommerce/google-listings-and-ads/pull/1161/files#diff-4a1398fd19d01009faac82dda4cde8189fa7eda015600cd48ea54199b6cbdd49R51-R55? They are not exactly the same: one is `countryCode`, and another is `countryCodes` with plural "s".

Reply:
You're right, my bad!